chatterbot/logic/mathematical_evaluation.py

```diff
@@ -77,9 +77,7 @@ def process(self, statement):
 
         # Use the result cached by the process method if it exists
         if input_text in self.cache:
-            cached_result = self.cache[input_text]
-            self.cache = {}
-            return cached_result
+            return self.cache[input_text]
```

Comment:
Correct. Although the same is true of the previous version. It was consuming additional memory until some calculation was redone.
A different version might use `weakref` or `functoolslru_cache` (in python3) which both would not suffer this drawback.
Anyway, feel free to close the PR if you are not comfortable with the change.